web/hooks/use-new-contract.ts

```diff
@@ -0,0 +1,285 @@
+import { User } from 'common/user'
+import { NewQuestionParams } from 'web/components/new-contract/new-contract-panel'
+import { usePersistentLocalState } from './use-persistent-local-state'
+import {
+  Contract,
+  MAX_DESCRIPTION_LENGTH,
+  NON_BETTING_OUTCOMES,
+  OutcomeType,
+  Visibility,
+} from 'common/contract'
+import { useEffect, useState } from 'react'
+import { getGroup } from 'web/lib/supabase/group'
+import { getAnte } from 'common/economy'
+import dayjs from 'dayjs'
+import { Group } from 'common/group'
+import { generateJSON } from '@tiptap/core'
+import { STONK_NO, STONK_YES } from 'common/stonk'
+import { extensions } from 'common/util/parse'
+import { useTextEditor } from 'web/components/widgets/editor'
+import { safeLocalStorage } from 'web/lib/util/local'
+import { createMarket } from 'web/lib/firebase/api'
+import { removeUndefinedProps } from 'common/util/object'
+import { track } from 'web/lib/service/analytics'
+
+const descriptionPlaceholder = 'Optional. Provide background info and details.'
+
+export const useNewContract = (
```

Comment:
I'm not sure about moving this to a separate file because it is so relevant to the first file.

Frankly, I would never have even split it into its own hook (stephen did that). I would put it directly into the component body like we do for most complex components.

If there are modular pieces that can be abstracted out, those could be hooks. But updating key state for a component seems like it should just be part of the component. And not at the bottom of the file lol, but in the component.

Reply:
oh great yeah I didn't understand why that was a think I think i'm just gonna put it back in the function then lol